### ISSUE TITLE
Proposal: Change keyboarding behavior for HierarchicalNavigationView expand/collapse

### ISSUE THREAD

**Issue by anawishnoff** [microsoft/microsoft-ui-xaml#2297]
# Proposal: Change keyboarding behavior for HierarchicalNavigationView

## Summary
Currently, users on a keyboard use the space or enter key to expand a flyout, and the esc key to collapse/close/exit the flyout. Instead, arrow keys should be used to expand and collapse flyout sub-menus in Left compact and Top modes. 

In top mode: the down arrow key will expand the flyout. The esc key will still be used to close/exit the flyout.

In left-compact mode: the left arrow key will expand the flyout. The right arrow key will close/exit the flyout. 

## Rationale
With HierarchicalNavigationView, there needs to be a difference between selecting a top-level parent item and expanding its sub-menu. If a top-level parent item is selectable, mouse users can click on the item to select it, or click on the right-aligned chevron icon to expand it without selecting it. With the current keyboarding in place, this distinction is not possible. Using space or enter keys, you cannot expand a menu without invoking it. 

## Open Questions
- Should the keyboarding behavior change for a NavigationView in left-expanded mode, to be more consistent with the other two modes? This mode does not involve a flyout, but should still provide a way to differentiate between invoking and expanding.

**Comment by mdtauk** [comment]
This behaviour would have to adapt to handle **Right to Left** scenarios, and how would this affect those using a gamepad?
